Demo/Ser_Des/ser_des.h

```diff
@@ -0,0 +1,104 @@
+#ifndef SER_DES_H
+#define SER_DES_H
+
+#include <iostream>
+#include <fcntl.h>
+#include <unistd.h>
+#include <sys/mman.h>
+#include <cstring>
+#include <cerrno>
+
+#include <type_traits>
+
+struct file {
+    int fd;
+
+    explicit file(const std::string& path)
+        : fd(open(path.c_str() , O_RDWR | O_CREAT , 0666)){}
+
+    void write_to_file(const int number)const {
```

Comment:
Class file should not provide functions such `read_from_file` and `write_to_file`.
The function names are not descriptive. The body reads and writes an int but the names have nothing to do with that.